src/Node/Stream.purs

```diff
@@ -61,6 +66,16 @@ type Writable r = Stream (write :: Write | r)
 -- | A duplex (readable _and_ writable stream)
 type Duplex = Stream (read :: Read, write :: Write)
 
+type Data = Either String Buffer
+
+-- | Read a chunk of data into `Data`
+-- | This is a mere internal plumbing function used by functions reading data
+-- | from a stream.
+type ChunkReader = Foreign -> Data
+readChunk :: ChunkReader
```

Comment:
Since this is just an internal function, I'm tempted to use a universally quantified type here instead of `Foreign` in order to avoid the dependency on `purescript-foreign`.